src/java/org/apache/cassandra/service/reads/ReplicaFilteringProtection.java

```diff
@@ -286,17 +294,25 @@ private void incrementCachedRows()
         if (currentRowsCached == cachedRowsFailThreshold + 1)
         {
             hitFailureThreshold = true;
-            String message = String.format("Replica filtering protection has cached %d rows during query %s, " +
-                                           "which is over the failure threshold of %d rows defined by " +
-                                           "'cached_replica_rows_fail_threshold' in cassandra.yaml.",
-                                           currentRowsCached, command.toCQLString(), cachedRowsFailThreshold);
-
-            logger.error(message);
-            Tracing.trace(message);
-            throw new OverloadedException(message);
+            String clearMessage = cachedRowsFailMessage(false);
+            String redactedMessage = cachedRowsFailMessage(true);
+
+            logger.error(redactedMessage);
+            Tracing.trace(clearMessage);
+            throw new OverloadedException(redactedMessage);
         }
     }
 
+    private String cachedRowsFailMessage(boolean redact)
+    {
+        return String.format("Replica filtering protection has cached %d rows during query %s, " +
+                             "which is over the failure threshold of %d rows defined by " +
+                             "'cached_replica_rows_fail_threshold' in cassandra.yaml.",
+                             currentRowsCached,
```

Comment:
The message is the same as above. The difference is in the first argument: `currentRowsCached` vs `maxRowsCached`, thus, I suggest to fix them into a single method with an additional argument. I hope I haven't missed more differences between them.

Reply:
Thank you for explaining the difference. Then I understand that they are separate :)